Tasks/DeployAzureResourceGroup/MachineGroupAgentExtensionManager.ts

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+
+    public deleteMGExtension(virtualMachine) {
+        try {
+            var operationParameters = new OperationParameters("uninstallation");
+            operationParameters.deferred = Q.defer<string>();
+            var vmName = virtualMachine["name"]
+            var resourceGroupName = this._taskParameters.resourceGroupName;
+            var extensionParameters = this._formExtensionParameters(virtualMachine, operationParameters.operation);
+            var extensionName = extensionParameters["extensionName"];
+            var callback = (error, result, request, response) => {
+                if (error) {
+                    operationParameters.deferred.reject("");
+                }
+                else {
+                    operationParameters.deferred.resolve("");
+                }
+            }
+            var extensionDeletionCallback = (error, result, request, response) => {
+                if (error) {
+                    this.log(tl.loc("SkipDeleteExtension", extensionName, vmName));
+                    operationParameters.deferred.resolve("");
+                }
+                else {
+                    this.log(tl.loc("DeleteExtension", extensionName, vmName));
+                    this._computeClient.virtualMachineExtensions.deleteMethod(resourceGroupName, vmName, extensionName, callback);
+                }
+            }
+            this._computeClient.virtualMachineExtensions.get(resourceGroupName, vmName, extensionName, extensionDeletionCallback);
+            return operationParameters.deferred.promise;
+        }
+        catch (exception) {
+            this.log(tl.loc("MGAgentOperationOnAllVMsFailed", operationParameters.operation, exception.message));
+            tl.setResult(tl.TaskResult.Failed, tl.loc("MGAgentOperationOnAllVMsFailed", operationParameters.operation, exception.message));
+            operationParameters.deferred.reject(tl.loc("MGAgentOperationOnAllVMsFailed", operationParameters.operation, exception.message));
+        }
+    }
+
+    private _settlePromise(operationParameters) {
+        if (operationParameters.failureCount + operationParameters.successCount == operationParameters.vmCount) {
+            if (operationParameters.failureCount > 0) {
+                this.log(tl.loc("MGAgentOperationOnAllVMsFailed", operationParameters.operation, ""));
+                operationParameters.deferred.reject(operationParameters.operation);
+            }
+            else {
+                if (operationParameters.vmCount !== 0) {
+                    this.log(tl.loc("MGAgentOperationOnAllVMsSucceeded", operationParameters.operation));
+                }
+                operationParameters.deferred.resolve(operationParameters.operation);
+            }
+        }
+    }
+
+    private _getExtensionDeletionCallback(operationParameters, resourceGroupName, extensionName, vmName, callback) {
```

Comment:
instead of directly passing operationParameters, pass this or use directly if it works